File: backend/src/main/java/com/schoolfinder/api/MetaController.java
package com.schoolfinder.api;

import com.schoolfinder.api.Dtos.MetaResponse;
import com.schoolfinder.config.AppProperties;
import com.schoolfinder.domain.SchoolCategory;
import java.util.Arrays;
import java.util.List;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/meta")
public class MetaController {

    private final AppProperties props;

    public MetaController(AppProperties props) {
        this.props = props;
    }

    @GetMapping
    public MetaResponse meta() {
        List<String> categories = Arrays.stream(SchoolCategory.values()).map(Enum::name).toList();
        return new MetaResponse(
            "UniMatch Cameroon",
            "Find Your Future University",
            props.getRegion().getCountry(),
            props.getRegion().getCurrency(),
            props.getRegion().getMapCenterLat(),
            props.getRegion().getMapCenterLng(),
            categories,
            props.getFirebase().isEnabled()
        );
    }
}

File: backend/src/main/kotlin/com/schoolfinder/api/MetaController.kt
package com.schoolfinder.api

import com.schoolfinder.config.AppProperties
import com.schoolfinder.domain.SchoolCategory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/meta")
class MetaController(private val props: AppProperties) {

    @GetMapping
    fun meta() = MetaResponse(
        appName = "UniMatch Cameroon",
        tagline = "Find Your Future University",
        country = props.region.country,
        currency = props.region.currency,
        mapCenterLat = props.region.mapCenterLat,
        mapCenterLng = props.region.mapCenterLng,
        categories = SchoolCategory.entries.map { it.name },
        firebaseEnabled = props.firebase.enabled,
    )
}
